src/GregClient/AuthProviders/IAuthProvider.cs

```diff
@@ -28,6 +29,8 @@ public interface IAuthProvider
     /// </summary>
     public interface IOAuth2AuthProvider : IAuthProvider
     {
+        //TODO ideally would not force the use of restsharp just to set auth.
+
```

Comment:
so we are still exposing RestSharp types ?

Reply:
yes, as stated in the PR description, I don't want to mess with auth in this PR.

Reply:
actually @pinzart90 I take that back, I think I'll give this a shot in the same PR to avoid needing to send 3 follow up prs as well...

Reply:
giving up on solving this for now. Now that we have an API exposing the access token the one use case I had in mind becomes a non issue, so I don't think it's worth it at the moment to spend any more time on this.